src/lighteval/tasks/lighteval_task.py

```diff
@@ -744,7 +729,7 @@ def create_requests_from_tasks(  # noqa: C901
                     # Constructing the requests
                     docs[TaskExampleId(cur_task_name, doc_id_seed)] = doc
                     reqs = task.construct_requests(doc, ctx, doc_id_seed, cur_task_name)
-                    for req_type in req_types:
-                        requests[req_type].extend(reqs[req_type])
+                    for req_type, reqs in reqs.items():
```

Comment:
overwritting `reqs` locally could lead to unexpected behavior. you could use `for req_type, req in reqs.items():` instead

Reply:
Ahh yeah that's so stupid, good spot :thumbsup: